Argo/Globals/Parser.swift

```diff
@@ -0,0 +1,29 @@
+public enum Parser<T> {
+  case Success(Box<T>)
+  case TypeMismatch(String)
+  case MissingKey(String)
+
+  public var value: T? {
+    switch self {
+    case let .Success(box): return box.value
+    default: return .None
+    }
+  }
+}
+
+public func typeMismatch<T>(type: String, object: JSON) -> Parser<T> {
```

Comment:
I don't think the user needs access to this. We should probably default to `internal`, instead of defaulting to `public`.